### ISSUE TITLE
Failing snapshot tests with `pytest tests/snapshot_tests/test_snapshots.py`?

### ISSUE THREAD

**Issue by TomJGooding** [Textualize/textual#4734]
This is weird issue that has been bugging me for a couple of weeks. Despite everything passing when I run all tests with `make test`, when I try running only the snapshot tests this results in failing tests relating to the command palette.

Can anyone else reproduce this?

```python
------------------------- snapshot report summary --------------------------
2 snapshots failed. 293 snapshots passed.
========================= short test summary info ==========================
FAILED tests/snapshot_tests/test_snapshots.py::test_command_palette - AssertionError: assert False
FAILED tests/snapshot_tests/test_snapshots.py::test_command_palette_discovery - AssertionError: assert False
=========== 2 failed, 293 passed, 1 warning in 88.05s (0:01:28) ===========
```

**Comment by darrenburns** [comment]
Yes - it seems to be an issue where if you run the command palette tests + unit tests it produces different output vs running them in isolation.

It's been driving me nuts the past couple of days because I have to run every single test and then run them again to generate new snapshots.

**Comment by TomJGooding** [comment]
Thanks Darren for confirming it isn't just me!

Initially I didn't spot  any visual difference until I looked a bit closer!

![image](https://github.com/user-attachments/assets/aec32e5b-d20b-4ba7-86c6-25af25a43c23)

I'm struggling to understand why the SVG fill would change depending on how you run the tests though!

**Comment by willmcgugan** [comment]
There is a very curious line here...
https://github.com/Textualize/textual/blob/main/src/textual/command.py#L389

Seems relevant

**Comment by darrenburns** [comment]
Yeah, I remember this issue was happening in the past, and adding that line seemed to workaround it. The issue seems to have returned though - the workaround is no longer working!

**Comment by darrenburns** [comment]
If I recall correctly, the original issue was the the segments for the emoji had a different "color" attribute in the output SVG. Hardcoding the color in the CSS for SearchIcon resulted in a constant color.

I'm guessing the recent change to SVGs to normalise them has broken the workaround here 🤷.

**Comment by darrenburns** [comment]
Do we also need to hardcode the background color of the SearchIcon? 🤠

**Comment by TomJGooding** [comment]
> I'm guessing the recent change to SVGs to normalise them has broken the workaround here 🤷.

I think more likely the recent changes to the OptionList and/or CommandPalette in #4667, maybe where the background changed to transparent?

From a quick `git bisect` this issue started with the snapshot update in 793fcbd.

**Comment by darrenburns** [comment]
Also works if we get rid of the emoji, which is always a good solution 😉

**Comment by willmcgugan** [comment]
Can you print out the list of segments for both paths, and compare?

I can think of no earthy reason why they might differ, tbh.

**Comment by darrenburns** [comment]
I was thinking this is an issue around global caches - for example an `@lru_cache` is global and will persist across tests.

Perhaps some background color is being computed from `transparent` in the first test, being stored in the cache, then the same cache is being hit in the second test?

@willmcgugan I checked the SVG in the past and it was just the background color which differs - you can also see the background color differing on the snapshot test output if you look closely.

**Comment by willmcgugan** [comment]
The @lru_cache is a sound theory. Easy to test. We could remove all the lru_caches.

**Comment by darrenburns** [comment]
Tried but no luck. Although there are also some inside Rich that I didn't remove.

**Comment by darrenburns** [comment]
@TomJGooding This is still an issue, but if you run the tests with `make test` now, it'll distribute the tests across processes and is MUCH faster, so it makes this issue a little less painful. There's also `make test-snapshot-update`.

**Comment by TomJGooding** [comment]
It looks like these snapshots were updated recently in dd13211, now running only the snapshot tests doesn't have this issue!

I'll leave it up to you whether you want to close this issue or try to investigate why this was happening.

**Comment by TomJGooding** [comment]
Closed as stale.
